### ISSUE TITLE
Session not created for the chrome version

### ISSUE THREAD

**Issue by wleionly** [sylvaindumont/docker-node-karma-protractor-chrome#20]
I tried to add protractor tests in pipeline,  and pulled the image weboaks/node-karma-protractor-chrome:latest in our test, but this test can not work. Is it the matter that chrome version does not meet the chromedriver version? I get the following error message:

[07:22:55] I/update - chromedriver: unzipping chromedriver_2.46.zip
[07:22:55] I/update - chromedriver: setting permissions to 0755 for /home/pipeline/df901498-b87a-49b6-8a8e-823651ba55a6/node_modules/protractor/node_modules/webdriver-manager/selenium/chromedriver_2.46
(node:491) [DEP0022] DeprecationWarning: os.tmpDir() is deprecated. Use os.tmpdir() instead.
[07:22:56] I/launcher - Running 1 instances of WebDriver
[07:22:56] I/direct - Using ChromeDriver directly...
[07:22:56] E/launcher - session not created: Chrome version must be between 71 and 75
  (Driver info: chromedriver=2.46.628388 (4a34a70827ac54148e092aafb70504c4ea7ae926),platform=Linux 4.4.0-139-generic x86_64)
[07:22:56] E/launcher - SessionNotCreatedError: session not created: Chrome version must be between 71 and 75
  (Driver info: chromedriver=2.46.628388 (4a34a70827ac54148e092aafb70504c4ea7ae926),platform=Linux 4.4.0-139-generic x86_64)
    at Object.checkLegacyResponse (/home/pipeline/df901498-b87a-49b6-8a8e-823651ba55a6/node_modules/selenium-webdriver/lib/error.js:546:15)
    at parseHttpResponse (/home/pipeline/df901498-b87a-49b6-8a8e-823651ba55a6/node_modules/selenium-webdriver/lib/http.js:509:13)
    at doSend.then.response (/home/pipeline/df901498-b87a-49b6-8a8e-823651ba55a6/node_modules/selenium-webdriver/lib/http.js:441:30)
    at process._tickCallback (internal/process/next_tick.js:68:7)
From: Task: WebDriver.createSession()
    at Function.createSession (/home/pipeline/df901498-b87a-49b6-8a8e-823651ba55a6/node_modules/selenium-webdriver/lib/webdriver.js:769:24)
    at Function.createSession (/home/pipeline/df901498-b87a-49b6-8a8e-823651ba55a6/node_modules/selenium-webdriver/chrome.js:761:15)

**Comment by wleionly** [comment]
I found the #19 issue, and I try to update webdriver to 2.43, and then it works. Chrome driver has release the ChromeDriver 73.0.3683.20, will docker-node-karma-protractor-chrome update?

**Comment by sylvaindumont** [comment]
@wleionly  Thanks for raising an issue, I just rebuilt the debian based images (latest,xvfb,headless,debian-*) images with latest available chromium version (72.0.3626.96). alpine based images are still on chromium 71.0.3578.98 until a new chromium build is released on alpine repos.

**Comment by ngeor-sendasyou** [comment]
@sylvaindumont we use the headless image in our builds, we started receiving the following exception today around an hour ago:

```
21 02 2019 14:17:16.107:INFO [launcher]: Starting browser ChromiumHeadless
21 02 2019 14:17:17.117:ERROR [launcher]: Cannot start ChromiumHeadless
	Received signal 11 SEGV_MAPERR 000000000080
#0 0x55e91361a711 <unknown>
#1 0x55e91361ab7b <unknown>
#2 0x55e91361b1de <unknown>
#3 0x7feecd8d40c0 <unknown>
#4 0x55e911a02314 <unknown>
#5 0x55e911a0d1b7 <unknown>
#6 0x55e9170f5566 <unknown>
#7 0x55e9170f561a <unknown>
#8 0x55e91199efd3 <unknown>
#9 0x55e911e18c32 <unknown>
#10 0x55e9119a1089 <unknown>
#11 0x55e9119a2039 <unknown>
#12 0x55e917bc7775 <unknown>
#13 0x55e9130c5257 <unknown>
#14 0x55e9130c5501 <unknown>
#15 0x55e9130c58b0 <unknown>
#16 0x55e9130d0b7a <unknown>
#17 0x55e9130c36c5 <unknown>
#18 0x55e9170fb2e3 <unknown>
#19 0x55e9170fb4d4 <unknown>
#20 0x55e9130cdd49 <unknown>
#21 0x55e910d6bd6d ChromeMain
#22 0x7feebfcfe2e1 __libc_start_main
#23 0x55e910d6bb8a _start
  r8: 0000000000000001  r9: 0000000000000020 r10: 00007fee957fa9d0 r11: 0000000000000202
 r12: 00007ffe50f60000 r13: 000055e91c303480 r14: 00007ffe50f60050 r15: 000055e91c303c10
  di: 00007ffe50f60000  si: 000055e9181b2770  bp: 00007ffe50f600a0  bx: 000055e91c303ce0
  dx: 000055e911a02314  ax: 00007ffe50f60000  cx: 0000000000000319  sp: 00007ffe50f5ffa0
  ip: 000055e911a02314 efl: 0000000000010206 cgf: 002b000000000033 erf: 0000000000000004
 trp: 000000000000000e msk: 0000000000000000 cr2: 0000000000000080
[end of stack trace]
Calling _exit(1). Core file will not be generated.
21 02 2019 14:17:17.117:ERROR [launcher]: ChromiumHeadless stdout: 
21 02 2019 14:17:17.117:ERROR [launcher]: ChromiumHeadless stderr: Received signal 11 SEGV_MAPERR 000000000080
#0 0x55e91361a711 <unknown>
#1 0x55e91361ab7b <unknown>
#2 0x55e91361b1de <unknown>
#3 0x7feecd8d40c0 <unknown>
#4 0x55e911a02314 <unknown>
#5 0x55e911a0d1b7 <unknown>
#6 0x55e9170f5566 <unknown>
#7 0x55e9170f561a <unknown>
#8 0x55e91199efd3 <unknown>
#9 0x55e911e18c32 <unknown>
#10 0x55e9119a1089 <unknown>
#11 0x55e9119a2039 <unknown>
#12 0x55e917bc7775 <unknown>
#13 0x55e9130c5257 <unknown>
#14 0x55e9130c5501 <unknown>
#15 0x55e9130c58b0 <unknown>
#16 0x55e9130d0b7a <unknown>
#17 0x55e9130c36c5 <unknown>
#18 0x55e9170fb2e3 <unknown>
#19 0x55e9170fb4d4 <unknown>
#20 0x55e9130cdd49 <unknown>
#21 0x55e910d6bd6d ChromeMain
#22 0x7feebfcfe2e1 __libc_start_main
#23 0x55e910d6bb8a _start
  r8: 0000000000000001  r9: 0000000000000020 r10: 00007fee957fa9d0 r11: 0000000000000202
 r12: 00007ffe50f60000 r13: 000055e91c303480 r14: 00007ffe50f60050 r15: 000055e91c303c10
  di: 00007ffe50f60000  si: 000055e9181b2770  bp: 00007ffe50f600a0  bx: 000055e91c303ce0
  dx: 000055e911a02314  ax: 00007ffe50f60000  cx: 0000000000000319  sp: 00007ffe50f5ffa0
  ip: 000055e911a02314 efl: 0000000000010206 cgf: 002b000000000033 erf: 0000000000000004
 trp: 000000000000000e msk: 0000000000000000 cr2: 0000000000000080
[end of stack trace]
Calling _exit(1). Core file will not be generated.
```

Do you have a different headless tag (older) that we could perhaps use?

**Comment by sylvaindumont** [comment]
@ngeor-sendasyou sorry for the faulty build, `latest` and `xvfb` are rebuilding with chromium 70, which doesn't segfault

**Comment by ngeor-sendasyou** [comment]
Just curious, how do you select a specific Chromium version when you build the image?

**Comment by sylvaindumont** [comment]
usually, it's just the latest version in `stretch/updates/main` with the latest version in `stretch/main` as a fallback

**Comment by ngeor-sendasyou** [comment]
Good morning @sylvaindumont ,
now we have a different problem:

```
[19:24:23] E/launcher - Error: SessionNotCreatedError: session not created: Chrome version must be between 71 and 75
```

I think that the previous headless tag was on Chrome 71.

I would like to suggest two things, but I don't know how easy it is:

1. make the headless tag use Chromium 71
2. to prevent overwriting the `headless` tag in the future, publish tags like `headless-70`, `headless-71`, etc

Would that be possible?

**Comment by wleionly** [comment]
Thanks @sylvaindumont

**Comment by ngeor-sendasyou** [comment]
I tried a bit to get an image with Chromium 71 but I can't seem to find it in Debian anymore. It seems it has 70 and 72. I did manage to find that Ubuntu Xenial is on Chromium 71, so I baked my own image which seems to work:  https://hub.docker.com/r/ngeor/node-chrome (the tag is 10-xenial-11)

So if the issue is fixed for @wleionly then I'm also fine with closing it.

Thank you @sylvaindumont
